src/actions/attack.cpp

```diff
@@ -316,8 +328,21 @@ battle_context_unit_stats::battle_context_unit_stats(const unit_type* u_type,
 
 	int base_damage = weapon->modified_damage(backstab_pos);
 	int damage_multiplier = 100;
-	damage_multiplier
-			+= generic_combat_modifier(lawful_bonus, u_type->alignment(), u_type->musthave_status("fearless"));
+	alignment = weapon->get_special_bool("alignment");
+
+	unit_ability_list alignment_specials = weapon->get_specials("alignment");
+		if(alignment) {
+                alignment_id=(*alignment_specials.front().first)["alignment"].str();
+                if(alignment_id.empty()) {
+                        alignment_id = (*alignment_specials.front().first)["id"].str();
+                }
+		damage_multiplier += special_combat_modifier(
+			lawful_bonus, alignment_id, u_type->alignment(), u_type->musthave_status("fearless"));
+            } else {
+	    damage_multiplier += generic_combat_modifier(lawful_bonus, u_type->alignment(), u_type->musthave_status("fearless"));
+	}
+
+
```

Comment:
This looks suspiciously like the change, above, so it looks like you wanted to make a function to avoid repeating yourself.